### PR TITLE
Simplify Input and output entries

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Part of https://github.com/restatedev/service-protocol/issues/60

### CODE REVIEW

File: crates/service-protocol/service-protocol/dev/restate/service/protocol.proto [hunk 1]
```diff
@@ -40,10 +40,8 @@ message StartMessage {
   repeated StateEntry state_map = 4;
   bool partial_state = 5;
 
-  repeated Header headers = 6;
-
   // If this invocation has a key associated (e.g. for objects and workflows), then this key is filled in. Empty otherwise.
-  string key = 7;
+  string key = 6;
```
Comment (tillrohrmann) [inline crates/service-protocol/service-protocol/dev/restate/service/protocol.proto:44]: I assume this warrants a protocol version bump if we haven't done it already compared to version `0.8`.

File: crates/service-protocol/src/codec.rs [hunk 2]
```diff
@@ -321,21 +319,10 @@ mod mocks {
...
-                        poll_input_stream_entry_message::Result::Value(value)
-                    }
-                    EntryResult::Failure(code, reason) => {
-                        poll_input_stream_entry_message::Result::Failure(Failure {
-                            code: code.into(),
-                            message: reason.to_string(),
-                        })
-                    }
-                }),
+        fn serialize_poll_input_stream_entry(InputEntry { value }: InputEntry) -> Bytes {
```
Comment (tillrohrmann) [inline crates/service-protocol/src/codec.rs:322]:
```suggestion
        fn serialize_input_entry(InputEntry { value }: InputEntry) -> Bytes {
```

**Review by tillrohrmann — APPROVED** [review]
Thanks for creating this PR @slinkydeveloper. The changes look good to me. Please bump the invoker protocol version if not already done. +1 for merging after making the e2e tests pass.

File: crates/service-protocol/service-protocol/dev/restate/service/protocol.proto (hunk 1 above)
Comment (slinkydeveloper) [reply crates/service-protocol/service-protocol/dev/restate/service/protocol.proto:44]: I was thinking to do this bump later, but ok will do it now.

**Pull request merged**